specs/DataSet.rst

```diff
@@ -0,0 +1,253 @@
+=====================
+DataSet Specification
+=====================
+
+Introduction
+============
+
+The DataSet class is used in QCoDeS to hold measurement results.
+It is the destination for measurement loops and the source for plotting and data analysis.
+As such, it is a central component of QCoDeS.
+
+The DataSet class should be usable on its own, without other QCoDeS components.
+In particular, the DataSet class should not require the use of Loop and parameters, although it should integrate with those components seamlessly.
+This will significantly improve the modularity of QCoDeS by allowing users to plug into and extend the package in many different ways.
+As long as a DataSet is used for data storage, users can freely select the QCoDeS components they want to use.
```

Comment:
Modularity should work both ways, when it makes sense. Specifically, modularity should respect layering: it should always be possible to use lower-level, more "base" components without higher-level components, but it may be acceptable to have higher-level components rely on base components. 

As an example, DataSet should be usable stand-alone, but Loop can require DataSet if that makes sense. An important corollary is that anything else that uses DataSet, like plotting, should work just as well if the DataSet is filled in by hand as when it is filled in using Loop.